lib/turn.rb

```diff
@@ -0,0 +1,80 @@
+class Turn
+  attr_reader :player1, :player2, :spoils_of_war
+
+  def initialize(player1, player2)
+    @player1 = player1
+    @player2 = player2
+    @spoils_of_war = []
+  end
+
+  def type
+    if player1.deck.rank_of_card_at(0) == player2.deck.rank_of_card_at(0) && player1.deck.rank_of_card_at(2) == player2.deck.rank_of_card_at(2)
+      :mutually_assured_distruction
+    elsif player1.deck.rank_of_card_at(0) == player2.deck.rank_of_card_at(0) &&  player1.deck.rank_of_card_at(2) != player2.deck.rank_of_card_at(2)
+      :war
+    else @player1.deck.rank_of_card_at(0) != @player2.deck.rank_of_card_at(0)
+      :basic
+    end
+  end
+
+  def winner
+  if type == :mutually_assured_distruction
+    "No Winner"
+  elsif type == :war
+    war_winner
+  else type == :basic
+      basic_winner
+    end
+  end
+
+  def basic_winner
+    if @player1.deck.rank_of_card_at(0) > @player2.deck.rank_of_card_at(0)
+       @player1
+    else
+      @player2
+    end
+  end
+
+  def war_winner
+    if player1.deck.rank_of_card_at(2) <= player2.deck.rank_of_card_at(2)
+      player2
+    else
+      player1
+    end
```

Comment:
```suggestion
    if @player1.deck.rank_of_card_at(2) <= @player2.deck.rank_of_card_at(2)
      @player2
    else
      @player1
    end
```

Though not explicitly necessary, it is convention to use the instance variable within the class where it is defined (rather than the method of the same name). It does work to use the method, just not preferred.

This is true pretty much anywhere you are calling `player1` or `player2` but I won't comment on every instance of it :)